core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -449,17 +449,19 @@ open class TileInfo {
 
     /** Shows important properties of this tile for debugging _only_, it helps to see what you're doing */
     override fun toString(): String {
-        val lineList = arrayListOf("TileInfo @($position)")
-        if (isCityCenter()) lineList += getCity()!!.name
-        lineList += baseTerrain
-        for (terrainFeature in terrainFeatures) lineList += terrainFeature
-        if (resource != null ) lineList += resource!!
-        if (naturalWonder != null) lineList += naturalWonder!!
-        if (roadStatus !== RoadStatus.None && !isCityCenter()) lineList += roadStatus.name
-        if (improvement != null) lineList += improvement!!
-        if (civilianUnit != null) lineList += civilianUnit!!.name + " - " + civilianUnit!!.civInfo.civName
-        if (militaryUnit != null) lineList += militaryUnit!!.name + " - " + militaryUnit!!.civInfo.civName
-        if (isImpassible()) lineList += Constants.impassable
+        val lineList = arrayListOf("TileInfo @$position")
+        if (this::baseTerrain.isInitialized) {
```

Comment:
Just return here if uninitialised, to exit early and minimize indentation
(I'm a big believer in 'exit early' if you hadn't noticed)

Reply:
True, and of course I did consider it and weighed having more indentation against an extra `joinToString` call (or `lineList[0] + " not yet ripe"`) the compiler might optimize away or may not... I draw the line at 4-5 levels of indentation in python or visual basic (yes, even that - in SSIS transforms) and maybe one more in curly braces sporting languages...  :partying_face:  - Thinking about compiler optimizations in code that only ever gets run by the debugger is a good one.

![image](https://user-images.githubusercontent.com/63000004/120351929-91e43300-c300-11eb-9b37-08146bbf1a26.png)
.. seems to work. Except it wants a merge... OK, time for experiments (aka 'does chromium have profiles'?).